src/utils/hooks/useAlerts/utils/alertQueries.ts

```diff
@@ -0,0 +1,48 @@
+import { KUBEVIRT } from '@kubevirt-utils/constants/constants';
+import { OPERATOR_LABEL_KEY } from '@kubevirt-utils/constants/prometheus';
+import { ALL_CLUSTERS_KEY } from '@kubevirt-utils/hooks/constants';
+import { OPERATOR_HEALTH_IMPACT_LABEL } from '@kubevirt-utils/hooks/useInfrastructureAlerts/utils/constants';
+import { escapePromLabelValue } from '@kubevirt-utils/utils/prometheus';
+import { AlertStates } from '@openshift-console/dynamic-plugin-sdk';
+
+import { OPERATOR_HEALTH_IMPACT } from './constants';
+
+type OperatorHealthImpact =
+  | typeof OPERATOR_HEALTH_IMPACT.NONE
+  | typeof OPERATOR_HEALTH_IMPACT.NOT_NONE;
+
+export const buildAlertsQuery = (
+  cluster: null | string,
+  hubClusterName: null | string,
+  operatorHealthImpact?: OperatorHealthImpact,
+  severity?: string,
+  selectedClusters?: string[],
+): string => {
+  const filters: string[] = [
+    `alertstate="${AlertStates.Firing}"`,
+    `${OPERATOR_LABEL_KEY}="${KUBEVIRT}"`,
+  ];
+
+  if (operatorHealthImpact === OPERATOR_HEALTH_IMPACT.NOT_NONE) {
+    filters.push(`${OPERATOR_HEALTH_IMPACT_LABEL}!="${OPERATOR_HEALTH_IMPACT.NONE}"`);
+  } else if (operatorHealthImpact === OPERATOR_HEALTH_IMPACT.NONE) {
+    filters.push(`${OPERATOR_HEALTH_IMPACT_LABEL}="${OPERATOR_HEALTH_IMPACT.NONE}"`);
+  }
+
+  if (severity) {
+    filters.push(`severity="${escapePromLabelValue(severity)}"`);
+  }
+
+  // Handle multiple selected clusters using regex match
+  if (selectedClusters && selectedClusters.length > 0) {
+    // Escape special regex characters and join with |
+    const escapedClusters = selectedClusters
+      .map((c) => c.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'))
```

Comment:
can we have a const for this REGEX? or a function for this replace. So we give a name and understand better what this do